File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int len = 0;
        ListNode* current = head;
        while (current != nullptr) {
            current = current->next;
            len++;
        }
        if (len == n) {
            head = head->next;
            return head;
        }

        current = head;
        int m = 0;

        while (current != nullptr) {
            m++;
            if ((len - m) == n) {

                current->next = current->next->next;
            }
            current = current->next;
        }

        return head;
    }
};

File: Data Structures & Algorithms/palindrome-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        current = head
        nodes = []

        while current:
            nodes.append(current)
            current = current.next

        start = 0
        end = len(nodes) - 1

        while start < end:
            if nodes[start].val != nodes[end].val:
                return False
            start += 1
            end -= 1
        return True
        